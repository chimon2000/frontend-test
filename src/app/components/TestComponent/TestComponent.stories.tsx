import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";

import { TestComponent } from "./TestComponent";

const stories = storiesOf("Components", module);

stories.add(
  "TestComponent",
  withInfo({ inline: true })(() => <TestComponent title="I work, yay!" />)
);
