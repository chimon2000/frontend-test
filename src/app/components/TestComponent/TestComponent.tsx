import * as React from "react";

export interface TestComponentProps {
  title: string;
}

export const TestComponent: React.FunctionComponent<TestComponentProps> = ({
  title
}) => <div>Test, {title}!</div>;
