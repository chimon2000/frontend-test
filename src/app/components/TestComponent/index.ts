export { TestComponent as default, TestComponentProps } from "./TestComponent";
