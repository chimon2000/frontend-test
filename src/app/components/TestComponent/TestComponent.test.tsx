import * as React from "react";
import { shallow } from "enzyme";

import { TestComponent } from "./TestComponent";

describe("<TestComponent />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(<TestComponent title="test" />);

    expect(wrapper).toMatchSnapshot();
  });
});
