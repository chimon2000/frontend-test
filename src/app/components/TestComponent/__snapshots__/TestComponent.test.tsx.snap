// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestComponent /> doesn't blow up 1`] = `
<div
  className="test-component"
>
  Test, 
  test
  !
</div>
`;
